### DISCUSSION TITLE
Reload new users without restarting strongswan

### DISCUSSION THREAD

**Discussion by realbitt** [strongswan/strongswan#785] [General]
Hi,
I have strongswan running on centos8 and everything is ok, now when i added new users to ipsec.secrets i want to reload new users without restarting the whole service to avoid disconnecting other connected users, i have tried **swanctl --load-creds** but not working.

**Comment by tobiasbrunner** [comment]
`ipsec.secrets` is a legacy config file that has nothing to do with `swanctl`.  Please see [ConfigurationFiles](https://wiki.strongswan.org/projects/strongswan/wiki/ConfigurationFiles).

**Reply by realbitt** [reply]
but ipsec not installed , and in my case swanctl working with ipsec.secrets and any users added to it after restarting strongswan i can able to connect with it.

i need to work with secrets file not putting users inside configuration file to make adding removing by automated service. 

What is the correct secrets file can work with swanctl ?  

[https://www.smashban.com/media/secrets.png](https://www.smashban.com/media/secrets.png)

**Reply by tobiasbrunner** [reply]
> but ipsec not installed

The script is renamed to `strongswan` on RPM-based systems. 

> and in my case swanctl working with ipsec.secrets

It's not, it's just the _stroke_ plugin loading `ipsec.secrets` when the daemon is started. That has nothing to do with `swanctl` or the _vici_ plugin.  To trigger a reload of that file, you'd have to use `strongswan rereadsecrets`, see [ipseccommand](https://wiki.strongswan.org/projects/strongswan/wiki/ipseccommand) for details.

> i need to work with secrets file not putting users inside configuration file to make adding removing by automated service.

You can do that with `swanctl.conf` too e.g. via `include` statements (possibly even with wildcards if you use a separate file for each user).

**Reply by realbitt** [reply]
Thank you for patience

`The script is renamed to strongswan on RPM-based systems.`
Wow,you really surprised me.

**Now**
Anyway, I'm a bit confused, I see now ipsec(strongswan) and swanctl working together since a lot of things I've done with swanctl and its working fine, is it case normal?

bear with me a little please

**strongswan rereadsecrets** not woking with me :(

`You can do that with swanctl.conf too e.g.`
How ?

**Reply by realbitt** [reply]
**strongswan stroke rereadsecrets**, works with me, thanks a lot.

now if u can help me to use **secrets** file with swanctl.conf (include) I will be grateful

**Reply by tobiasbrunner** [reply]
> Anyway, I'm a bit confused, I see now ipsec(strongswan) and swanctl working together since a lot of things I've done with swanctl and its working fine, is it case normal?

Yeah, sure, they can work together, but as you already noticed it could be confusing to mix the configuration backends.

> **strongswan stroke rereadsecrets**, works with me, thanks a lot.

That should be the same as `strongswan rereadsecrets` (the script literally just converts it to a `/path/to/stroke rereadsecrets` call).

> now if u can help me to use **secrets** file with swanctl.conf (include) I will be grateful

Just put e.g.

    secrets {
      include users/*.conf
    }

at the end of `swanctl.conf`, then create a `users` directory in the `swanctl` directory (where `swanctl.conf` is located) and then for each user create a `<username|userid|whatever>.conf` file with something like this:

    eap-<username|userid|whatever> {
      id = <username>
      secret = <password>
    }

(Note that the section name `eap-...` has to be unique and can't contain any dots.)

**Reply by realbitt** [reply]
thank you very mush, legend man ;)

**Comment by ghost** [comment]
In Ubuntu 20.04 and ArchLinux I use `ipsec rereadsecrets`. Maybe, it will work in CentOS?

**Reply by realbitt** [reply]
@Morion-Self 
yes works fine, but now i moved to radius its perfect and no need to read/load users every new account .
